### DISCUSSION TITLE
Polish Translation (invitation to check + posted translation notes)

### DISCUSSION THREAD

**Discussion by EvidentlyCube** [TiddlyWiki/TiddlyWiki5#6080] [General]
I've made a Polish translation plugin (See #6079). You can [check it here|https://cdn.evidentlycube.com/prototypes/tiddlywiki-pl/index.html]. Notes about that:

 - I couldn't figure out how to build an empty TW with the language plugin bundled in, so you get the full build instead
 - I also couldn't figure out how to make it start in Polish so you need to change the language manually

If there are any people who know Polish any feedback will be appreciated :).

I'll also paste translation notes in a reply, just to keep them for posterity since they may be useful in the future.

**Comment by EvidentlyCube** [comment]
# Polish Translation Notes

## Translations
 
| English            | Polish                           |
| ------------------ | -------------------------------- |
| **alert**          | `alert`                          |
| **bitmap editor**  | `edytor obrazów`                 |
| **deserializer**   | `deserializer`                   |
| **draft**          | `szkic`                          |
| **macro**          | `makro`                          |
| **muted**          | `stonowany`                      |
| **operand**        | `argument`                       |
| **parser**         | `parser`                         |
| **pattern**        | `wzorzec`                        |
| **shadow tiddler** | `tiddler-cień`                |
| **story river**    | `Story River` (make it capital!) |
| **story view**     | `Widok Story River`              |
| **tiddler**        | `tiddler`                        |
| **toolbar**        | `pasek narzędzi`                 |
| **wizard**         | `okno modalne`                   |


## Explanations:

Note: the original translation was done by a native speaker who spends most of their time using English when working with web, tool interfaces and programming, as such sometimes I had to use google to find how certain things are most commonly called in Polish books and Internet. 


### alert

While this could also be translated to "okno podręczne", I got the impression "alert" is as, if not more, popular in Polish internet, so I went with that one.

### deserializer, parser

I kept using the original English phrases, because those are used in tools for power users, and those can be expected to either have minimal knowledge about programming or be able to look the information up easily.


### story river / view

"Story River" refers to the section of the page which displays opened tiddlers. Looking at how other languages tackled the translation I noticed a two trends: keep using the English phrase (German, Dutch, Slovak) or translate it directly (or slightly differently). Since it's a phrase invented for TiddlyWiki, I've decided to keep the original English phrase instead of trying to translate it. This will both avoid any awkward, word-salad-like phrasing, as well as simplify understanding English documentation or plugins.

And by extension "Story View" became "Widok Story River", since in essence it's just configfuration of how Story River is displayed.

### shadow tiddler

~~I considered going for "tiddler wtyczki" (plugin tiddler) but "ukryty tiddler" (hidden tiddler) just seems to roll off the tongue better.~~

After a conversation in this thread I went for `tiddler-cień`, since that does say the tiniest bit about its nature (`cień` suggesting it's somehow hidden or _subordinate_ to something else), while also being unclear enough that it should prompt the user to learn more about their nature without assuming anything. Also for people who want to search for help in English or use English plugins it'll be easier to map this to "shadow tiddler" than the original `ukryty tiddler`.

**Reply by pmario** [reply]
> **shadow tiddler**
>
>I considered going for "tiddler wtyczki" (plugin tiddler) but "ukryty tiddler" (hidden tiddler) just seems to roll off the tongue better.

hmm. system tiddlers are also "hidden". ... Every plutin-tiddler is a shadow-tiddler. They can be overwritten by the user and will win, when the wiki is started. If a shadow is overwritten it becomes a system-tiddler. 

If a system-tiddler is deleted and a shadow-tiddler with the same name exists, the shadow-tiddler takes over again. ... 

So I'm not 100% sure if it should mean "hidden tiddler" ... BUT ... If it feels right for a native speaker I would be OK with it ;)

**Reply by pmario** [reply]
A "plugin-tiddler" in tiddlywiki is a tiddler that contains a plugin. A plugin can contain several shadow-tiddlers, that are "expanded" on wiki startup.

**Reply by pmario** [reply]
We had to invent a lot of new terms for tiddlers, because they are like a chameleon.

**Reply by EvidentlyCube** [reply]
System tiddlers are translated to "tiddler systemowy" (which is a direct translation). Unfortunately anything that involves the Polish word for shadow would just make it sound like some kind of magical tiddler from a fantasy movie/DnD campaign. Which is why I tried to look for something semantically similar - "shadow" implies something is less visible, and what's less visible? hidden things!

I agree there is more nuance in "shadow tiddler" than in "hidden tiddler" but I've not been able to find a adjective that could show the nuance but at the same time be suggesting on its own of the behavior of the tiddler (not being visible). One good thing is "ukryty tiddler" certainly does sound natural :).

Thanks for the feedback!

**Comment by pmario** [comment]
>Unfortunately anything that involves the Polish word for shadow would just make it sound like some kind of magical tiddler from a fantasy movie/DnD campaign.

Yea, that's probably the right thing. It is "kind of" magical and imo it's one of TidddlyWiki's [USPs](https://en.wikipedia.org/wiki/Unique_selling_proposition). ... 

Since a shadow tiddler is "waiting" in the background until it is revoked again may be "**stand-by** tiddler" would be an option? It is different to system and hidden but imo similar enough to hidden to make sense. ... At least in English ;)

**Reply by EvidentlyCube** [reply]
Let's hope I am not overanalyzing things here :D.

----

So just to make sure I understand this correctly. Just fyi, here are : 
 
* [Google Group: Tweaking Shadow Filters](https://groups.google.com/g/tiddlywiki/c/T-nZkz7Qh-c)
*  [Google Group: Editing shadow filter](https://groups.google.com/g/tiddlywiki/c/A7xJk4RFOr8)
*  [Docs: Shadow Tiddlers](https://tiddlywiki.com/static/ShadowTiddlers.html)
*  [Dev Docs: Everything is a tiddler](https://tiddlywiki.com/dev/#Module%20System)
*  [Docs: System tiddlers](https://tiddlywiki.com/static/SystemTiddlers.html)

**Here is how I understand things:**
 
* System tiddlers load core system functionality and are loaded before anything else. They **can** be overriden by creating a tiddler with the same name, in essence causing them to become functionally equivalent with shadow filters (in that they are still stored but are inactive until the overriding tiddler is removed)
* Shadow tiddlers are _sub-tiddlers_ of a plugin. That is, a plugin consists of a plugin tiddler and any number of additional tiddlers - the additional ones are shadow tiddlers.
* If you create a tiddler with the same name as an existing shadow tiddler, it takes the shadow tiddler's place in the system. The original one is still stored in memory, but it won't be doing anything until you remove the overriding one.
* There is no specific word for a shadow tiddler that's active and a shadow tiddler that is overridden.

In that sense, the word "shadow" seems to have been used because of the mental image of the shadow being a bit like the thing that casts it, but hidden behind and passive. 

That gives me a bit more to work with, so here are some options:

 1. In "Shadow Tiddler" the word _shadow_ functions a bit like an adjective, and any way to do that in Polish (`Cienisty Tiddler`, `Tiddler Cienia`) sounds like something from a fantasy RPG, kinda distracting and with negative connotations.
 2.  `tiddler-cień` (literally "tiddler-shadow") is a decent option. It's two nouns, but it means that the tiddler is also a shadow at the same time. Unlike the two options from 1, this feels much less mystical and is more _grounded in reality_ so to speak. I'd stick to using the dash to very clearly indicate those two must be read as a single word.
 3.  `ukryty tiddler` (literally "hidden tiddler") the option I went for initially. It does make sense in certain aspects: shadows do tend to be hidden; the shadow tiddlers are hidden. But it can indeed give an incorrect impression about it functionality - or rather, it might cause the user to think it does less than it really does, causing them to be confused.
 4. `pod-tiddler` (literally "sub-tiddler") . That one also makes sense, because functionally they are sub-tiddlers of a plugin tiddler. It gives no suggestion about the overridability functionality though,
 5. `subtiddler` or `sub-tiddler`. The prefix `sub-` is used in some loan words, and while it _could_ work, it looks like a bad misspelling of "subtitle"
 6. Some ideas I considered but they just don't seem to work: `zapasowy tiddler` (backup/stand-by tiddler), `zamienny tiddler` (interchangeable tiddler), `dodatkowy tiddler` (additional tiddler). All of those don't sound right to me.

So after doing this deep dive, I think `tiddler-cień` is the best option, meaning I went full circle :).

**Reply by Jermolene** [reply]
Thanks @EvidentlyCube 

> **Here is how I understand things:**
> 
> * System tiddlers load core system functionality and are loaded before anything else. They **can** be overriden by creating a tiddler with the same name, in essence causing them to become functionally equivalent with shadow filters (in that they are still stored but are inactive until the overriding tiddler is removed)

Not quite right. System tiddlers are any tiddler whose title starts with the characters `$:/`. The only difference from ordinary tiddlers is that system tiddlers are hidden from most lists in the user interface (eg, they don't show up in "Recent" changes). The idea is to stop internal system components from cluttering up the user interface.

> * Shadow tiddlers are _sub-tiddlers_ of a plugin. That is, a plugin consists of a plugin tiddler and any number of additional tiddlers - the additional ones are shadow tiddlers.

Perhaps clearer to think about it like this: A plugin is a special type of tiddler that makes available any number of shadow tiddlers.

> * If you create a tiddler with the same name as an existing shadow tiddler, it takes the shadow tiddler's place in the system. The original one is still stored in memory, but it won't be doing anything until you remove the overriding one.

That's right.

> * There is no specific word for a shadow tiddler that's active and a shadow tiddler that is overridden.

We talk of "overridden shadow tiddlers" and occasionally "non-overridden shadow tiddlers".
 
> In that sense, the word "shadow" seems to have been used because of the mental image of the shadow being a bit like the thing that casts it, but hidden behind and passive.

That's right. In English we also use the word "shadow" to mean "an inseparable attendant or companion". For example, "Jim shadowed Bob for the rest of the fortnight to learn how to operate the machine".

Other words that are not synonyms but might have worked as alternatives include: phantom, ghost, spectre.

> 2. `tiddler-cień` (literally "tiddler-shadow") is a decent option. It's two nouns, but it means that the tiddler is also a shadow at the same time. Unlike the two options from 1, this feels much less mystical and is more _grounded in reality_ so to speak. I'd stick to using the dash to very clearly indicate those two must be read as a single word.
> So after doing this deep dive, I think `tiddler-cień` is the best option, meaning I went full circle :).

Great, that makes sense.

**Comment by Jermolene** [comment]
Thanks @EvidentlyCube. I tweeted about this discussion just in case there's others out there who might be able to help:

https://twitter.com/TiddlyWiki/status/1444224558345723905
